### ISSUE TITLE
Exception on windows node 0.12 - bad path handling

### ISSUE THREAD

**Issue by lukeapage** [rollup/rollup#119]
Hi,

Trying out this project and I get...
````
TypeError: Cannot read property '0' of null
    at Object.defaultExternalResolver [as resolveExternal] (node_modules\rollup\dist\rollup.js:2692:40)
    at Bundle.defaultResolver [as resolveId] (node_modules\rollup\dist\rollup.js:2684:18)
    at node_modules\rollup\dist\rollup.js:2947:42
    at Array.map (native)
    at Bundle.fetchAllDependencies (node_modules\rollup\dist\rollup.js:2946:38)
    at node_modules\rollup\dist\rollup.js:2937:19
    at lib$es6$promise$$internal$$tryCatch (node_modules\es6-promise\dist\es6-promise.js:331:16)
    at lib$es6$promise$$internal$$invokeCallback (node_modules\es6-promise\dist\es6-promise.js:343:17)
    at node_modules\es6-promise\dist\es6-promise.js:891:13
    at lib$es6$promise$asap$$flush (node_modules\es6-promise\dist\es6-promise.js:125:9)
```
for some reason my paths are being resolved by node to `c:/a/b/` and therefore your regex doesn't match...
```
"c:/a/b/c.js".match(/^(?:\/|(?:[A-Za-z]:)?\\)/)
```

Notice that after I fixed this, it was stuck in an infinite loop with the dirctory set to "." so other directory functions might need fixing too.

**Comment by lukeapage** [comment]
partial fix in #117

**Comment by Rich-Harris** [comment]
closing, #117 is merged
